database/ffldb/blockio.go

```diff
@@ -553,6 +575,38 @@ func (s *blockStore) readBlock(hash *chainhash.Hash, loc blockLocation) ([]byte,
 	return serializedData[8 : n-4], nil
 }
 
+// deleteBlocks will delete all blocks in the blockstore where the height of the
+// last block in the file is less than the provided height.
+func (s *blockStore) deleteBlocks(deleteBefore uint32) error {
+	s.fbhMutex.RLock()
```

Comment:
I think it is simpler to just take the `s.fbhMutex.Lock()` here instead and  `defer s.fbhMutex.Unlock()` for clarity rather than use both the read and write locks for this function.